apis/flinkcluster/v1beta1/flinkcluster_validate.go

```diff
@@ -259,6 +259,11 @@ func (v *Validator) validateJobUpdate(old *FlinkCluster, new *FlinkCluster) erro
 		newJobSpec, _ := json.Marshal(new.Spec.Job)
 		return fmt.Errorf("you cannot change cluster type between session cluster and job cluster, old spec.job: %q, new spec.job: %q", oldJobSpec, newJobSpec)
 	case old.Spec.Job.SavepointsDir == nil || *old.Spec.Job.SavepointsDir == "":
+		oldSpec, _ := json.Marshal(old.Spec)
```

Comment:
@live-wire what I think we should do here is to update `checkCancelRequested` func to check for the cancel control annotation as well; If it's a cancel request, we don't need to validate other stuff.

Reply:
https://github.com/spotify/flink-on-k8s-operator/blob/4729305be36f95c141737d6b43d6b65124741435/apis/flinkcluster/v1beta1/flinkcluster_validate.go#L173

Reply:
Hey @regadas 
Added a check to skip further validations if `job-cancel` is requested. 
Wdyt?